File: java/src/main/java/factory_method/product/SuperTruck.java
package factory_method.product;

import factory_method.product.part.Engine;
import factory_method.product.part.Tire;

public class SuperTruck extends Vehicle {
    public static final String NAME = "Super Truck";

    public SuperTruck() {
        name = NAME;
        engine = Engine.GASOLINE;
        tire = Tire.OFFLOAD;
    }
}

File: kotlin/src/main/kotlin/factory_method/product/SuperTruck.kt
package factory_method.product

import factory_method.product.part.Engine
import factory_method.product.part.Tire

class SuperTruck : Vehicle() {
    override var name = NAME
    override var engine: Engine = Engine.GASOLINE
    override var tire: Tire = Tire.OFFLOAD

    companion object {
        const val NAME: String = "Super Truck"
    }
}